dojo/metrics/views.py

```diff
@@ -698,6 +698,156 @@ def product_type_counts(request):
                   )
 
 
+def product_tag_counts(request):
+    form = ProductTagCountsForm()
+    opened_in_period_list = []
+    oip = None
+    cip = None
+    aip = None
+    all_current_in_pt = None
+    top_ten = None
+    pt = None
+    today = timezone.now()
+    first_of_month = today.replace(day=1, hour=0, minute=0, second=0, microsecond=0)
+    mid_month = first_of_month.replace(day=15, hour=23, minute=59, second=59, microsecond=999999)
+    end_of_month = mid_month.replace(day=monthrange(today.year, today.month)[1], hour=23, minute=59, second=59,
+                                     microsecond=999999)
+    start_date = first_of_month
+    end_date = end_of_month
+
+    if request.method == 'GET' and 'month' in request.GET and 'year' in request.GET and 'product_tag' in request.GET:
+        form = ProductTagCountsForm(request.GET)
+        if form.is_valid():
+            pt = form.cleaned_data['product_tag']
+            user_has_permission_or_403(request.user, pt, Permissions.Product_Type_View)  # FIXME
```

Comment:
Looks like this results in a 500 error when a user without proper permissions attempts to view this report. Any thoughts @Maffooch ?

```
django-defectdojo-uwsgi-1         | [22/Dec/2023 23:22:52] INFO [django.request:241] OK: /metrics/product/tag/counts
django-defectdojo-uwsgi-1         | Traceback (most recent call last):
django-defectdojo-uwsgi-1         |   File "/usr/local/lib/python3.11/site-packages/django/core/handlers/exception.py", line 56, in inner
django-defectdojo-uwsgi-1         |     response = get_response(request)
django-defectdojo-uwsgi-1         |                ^^^^^^^^^^^^^^^^^^^^^
django-defectdojo-uwsgi-1         |   File "/usr/local/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
django-defectdojo-uwsgi-1         |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
django-defectdojo-uwsgi-1         |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-defectdojo-uwsgi-1         |   File "/app/dojo/metrics/views.py", line 722, in product_tag_counts
django-defectdojo-uwsgi-1         |     user_has_permission_or_403(request.user, pt, Permissions.Product_Type_View)  # FIXME
django-defectdojo-uwsgi-1         |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-defectdojo-uwsgi-1         |   File "/app/dojo/authorization/authorization.py", line 250, in user_has_permission_or_403
django-defectdojo-uwsgi-1         |     if not user_has_permission(user, obj, permission):
django-defectdojo-uwsgi-1         |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-defectdojo-uwsgi-1         |   File "/app/dojo/authorization/authorization.py", line 205, in user_has_permission
django-defectdojo-uwsgi-1         |     raise NoAuthorizationImplementedError(
django-defectdojo-uwsgi-1         | dojo.authorization.authorization.NoAuthorizationImplementedError: No authorization implemented for class Tagulous_Product_tags and permission 1002
```
![Screenshot 2023-12-22 at 17 22 59](https://github.com/DefectDojo/django-DefectDojo/assets/1749665/cbeae043-951d-4eb1-b559-4b59424b64ad)

Reply:
Thanks for doing this testing @cneill it has uncovered two issues (that I can see atm)
- The queryset of tags passed to the filter needs to the set of tags across all products that user has access to
- When checking for authorization, the tag cannot be used here since there are not any authorization models on the tag level. Instead, I think you will have to get the set of findings a user has access to see (user the `get_authorized_finding` function and search the code for how it is used) and then filter those down by the product tag

Reply:
These should be fixed now (as described in the comment below).